iothub/device/src/Message.cs

```diff
@@ -10,14 +10,20 @@
 
 namespace Microsoft.Azure.Devices.Client
 {
+    internal enum StreamDisposalOwnership
```

Comment:
Might use the word "lifetime" instead of disposal, or the word "responsibility" instead of ownership.

So `StreamLifetimeOwnership` or `StreamDisposalResponsibility`.